src/Command/PurgeCommand.php

```diff
@@ -0,0 +1,359 @@
+            );
+
+            foreach ($rows as $row) {
+                $leadsData[$row['id']] = $row;
+            }
+        }
+
+
+        return $leadsData;
+    }
+
+    /**
+     * @param array $leadsData
+     * @param array $masterForm
+     * @return int
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    private function purgeLeadsData(array $leadsData, array $masterForm)
+    {
+        $deleted = 0;
+
+        if (!empty($leadsData)) {
+
+            $ids = implode(',', array_keys($leadsData));
+
+            $deleted = $this->db->executeUpdate(
+                "DELETE FROM tl_lead_data WHERE id IN(".$ids.")"
+            );
+
+            $logLevel = LogLevel::INFO;
+            $logMessage = 'Purged '.(int)$deleted.' leads data for master form "'.$masterForm['title'].'"';
+            $this->logger->log($logLevel, $logMessage, array('contao' => new ContaoContext(__METHOD__, $logLevel)));
+        }
+
+        return $deleted;
+    }
+
+    /**
+     * @param array $leadsData
+     * @param string $leadPurgeUploads
+     * @return array
+     */
+    private function getUploads(array $leadsData, string $leadPurgeUploads)
+    {
+        $uploads = [];
+        if (!empty($leadsData) && !empty($leadPurgeUploads)) {
+            foreach ($leadsData as $data) {
+                if ('upload' === $data['field_type']) {
+                    $uploads[$data['id']] = $this->getUploadFileModel($data['value']);
+                }
+            }
+        }
+
+        return $uploads;
+    }
+
+    /**
+     * @param string $value
+     * @return FilesModel|null
+     */
+    private function getUploadFileModel(string $value)
+    {
+        if (!Validator::isUuid($value)) {
+            return null;
+        }
+
+        $filesModel = FilesModel::findByUUid($value);
+
+        return $filesModel;
+    }
+
+
+    /**
+     * @param array $uploads
+     * @param array $masterForm
+     * @return int
+     */
+    private function purgeUploads(array $uploads, array $masterForm)
+    {
+        if (empty($uploads)) {
+            return 0;
+        }
+
+        $count = 0;
+        $deleted = 0;
+        foreach ($uploads as $dataId => $filesModel) {
+            $deleted += $this->purgeUpload($dataId, $filesModel);
+            $count++;
+        }
+
+        $logLevel = LogLevel::INFO;
+        $logMessage = 'Purged '.$deleted.' of '.$count.' leads uploads for master form "'.$masterForm['title'].'"';
+        $this->logger->log($logLevel, $logMessage, array('contao' => new ContaoContext(__METHOD__, $logLevel)));
+
+        return $deleted;
+    }
+
+    /**
+     * @param int $dataId
+     * @param FilesModel|null $filesModel
+     * @return int
+     */
+    private function purgeUpload(int $dataId, ?FilesModel $filesModel)
+    {
+        $count = 0;
+        $logLevel = LogLevel::ERROR;
+        $logMessage = 'Purge leads upload data id "'.$dataId.'": ';
+
+        if (null !== $filesModel) {
```

Comment:
You should generally try to use _early exits_ to prevent nested if-clauses. This could could look like this:

```php
if (null === $filesModel) {
    $this->logger->error('Model not found for deletion');
    return;
}

// ... continue with try-catch
```